benchmarking/evaluations/metrics/utils/student_calculations.py

```diff
@@ -9,7 +9,7 @@ def num_friends_satisfied(student: Student) -> int:
         if relation != Relationship.FRIEND:
             continue
         for teammate in student.team.students:
```

Comment:
good catch here, tbh I would do something like:
```py
if relation_student_id in [t.id for t in student.team.students if t.id != student.id]:
    count += 1
```

instead of this weird loop (also rename `student_id, relation` to `relation_student_id, relation`

Reply:
what do you think? @justino599 
Mostly because we actually do not want to count a friend as mattering if it's a student being friends with themselves